### ISSUE TITLE
Pretty-printing of `let%test _ = ...`

### ISSUE THREAD

**Issue by Kakadu** [camlp5/camlp5#89]
I got a few issues when using ppx_inline_tests and camlp5. Let's look at file demo1.ml.

> [%%test let _ = true]
> [%%test true]

In this file 1st line is correct ppx inline test, 2nd one -- doesn't. Let's preprocess it with camlp5

> ➜   camlp5o pr_dump.cmo -- demos/demo1.ml > b.ml
> ➜   ocamlc -c -stop-after parsing -dsource b.ml
> [%%test true]
> [%%test true]

It looks like pretty-printing of structure items is slightly wrong: it puts MLast.StExp  constructors instead of StVal with wildcards. Could you double check, @chetmurthy?

**Comment by chetmurthy** [comment]
1. I checked that what you're seeing is real: camlp5 8.00.XX does this, and I see in the code of (pa_o.ml) where it happens
2. But it turns out, camlp5 has done this as long ago as release 6.00 (so,  2010) and that code was still there.
3. here's the line from rel600: https://github.com/camlp5/camlp5/blob/f7dcaf1cc5288fb781df3f6e85ece1e3b5b16f76/etc/pa_o.ml#L413
4. and here it is from master: https://github.com/camlp5/camlp5/blob/a7bd520ae3ab666a06a124e22122ff6fbe14db47/etc/pa_o.ml#L1012

In short, I think this is "working as designed", where "designed" is "according to the idea of parsing that Daniel originally had".  I'm open to changing this, but it seems like it should only be for reasons that are functionally important.

**Comment by Kakadu** [comment]
Well, right this pretty-printing shortcoming damages preprocessing by camlp5 before preprocessing by ppx_imline_tests, because changes in AST described above. You could try the demo here: https://github.com/Kakadu/playground_JSX/tree/d1ba4af823d2ada476395cdb78dc75970d234d1c Running `dune test` works file, but if you uncomment a line in `demos/dune` you would see an error.

**Comment by chetmurthy** [comment]
Does ```let __ = true``` not work ? (two underscores, not one)

**Comment by Kakadu** [comment]
Ah, I found that we should write `let%test "name" = ...` and now it is pretty-printed as expected. It looks like my future tests are going to always have a name. Maybe it is a good thing *grin*
